frontend/src/App.tsx

```diff
@@ -1,66 +1,109 @@
+      checkConnection();
+      setLoaded(true);
+    }, 200)
+    if (connected) getCount();
+  }, [connected])
+ 
+  async function connect() {
+    if (window.fuel) {
+      try {
+        await window.fuel.connect();
+        const [account] = await window.fuel.accounts();
+        setAccount(account);
+        setConnected(true);
+      } catch (err) {
+        console.log("error connecting: ", err);
+      }
     }
-    main();
-  }, []);
-
+  }
+ 
+  async function checkConnection() {
+    if (window.fuel) {
+      const isConnected = await window.fuel.isConnected();
+      if (isConnected) {
+        const [account] = await window.fuel.accounts();
+        setAccount(account);
+        setConnected(true);
+      }
+    }
+  }
+ 
+  async function getCount() {
+    if (window.fuel) {
+      const wallet = await window.fuel.getWallet(account);
+      const contract = CounterContractAbi__factory.connect(CONTRACT_ID, wallet);
+      const { value } = await contract.functions.count().simulate();
+      setCounter(value.toNumber());
+    }
+  }
+ 
   async function increment() {
-    // a loading state
-    setLoading(true);
-    // Creates a transactions to call the increment function
-    // because it creates a TX and updates the contract state this requires the wallet to have enough coins to cover the costs and also to sign the Transaction
-    try {
-      await contract.functions.increment().txParams({ gasPrice: 1 }).call();
-      const { value } = await contract.functions.count().get();
-      setCounter(Number(value));
-    } finally {
-      setLoading(false);
+    if (window.fuel) {
+      const wallet = await window.fuel.getWallet(account);
+      const contract = CounterContractAbi__factory.connect(CONTRACT_ID, wallet);
+      // Creates a transactions to call the increment function
+      // because it creates a TX and updates the contract state this requires the wallet to have enough coins to cover the costs and also to sign the Transaction
+      try {
+        await contract.functions.increment().txParams({ gasPrice: 1 }).call();
+        getCount();
+      } catch (err) {
+        console.log("error sending transaction...", err);
+      }
     }
   }
-  
+ 
+  if (!loaded) return null
+ 
   return (
-    <div className="App">
-      <header className="App-header">
-        <p>Counter: {counter}</p>
-        <button disabled={loading} onClick={increment}>
-          {loading ? "Incrementing..." : "Increment"}
-        </button>
-      </header>
-    </div>
+    <>
+      <div className="App">
+        {
+          connected ? (
+            <>
+              <h3>Counter: {counter}</h3>
+              <button style={buttonStyle} onClick={increment}>
+                Increment
+              </button>
+            </>
+          ) : (
+            <button style={buttonStyle} onClick={connect}>Connect</button>
+          )
+        }
+      </div>
+    </>
   );
 }
+ 
 export default App;
+ 
+const buttonStyle = {
```

Comment:
Personally I prefer the styling of the app before. If we keep it, we don't need to update any screenshots in the docs :)

Reply:
This code comes from [this pr](https://github.com/FuelLabs/fuel-docs/pull/87) & I've tried to keep as it is to avoid breaking stuff before the release (I might better change an screenshot than a code).

However I think the quickstart needs a revision (not only the styling but cra itself and the step by step explanation).

Right now this codebase is replicated in 3 places so it can be messy to keep it synced. But I'm completely down to revisit and upgrade the code. 

@sarahschwartz what do you think?

Reply:
I agree, I think we should keep it the same for now, and make these updates in the next beta release.